multicluster/controllers/multicluster/leader/labelidentity_export_controller_test.go

```diff
@@ -185,6 +189,52 @@ func TestLabelIdentityResourceExportReconclie(t *testing.T) {
 	}
 }
 
+func TestConcurrentProcessLabelForResourceImport(t *testing.T) {
+	existingResImp := &mcsv1alpha1.ResourceImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: common.LeaderNamespace,
+			Name:      labelHash3,
+		},
+		Spec: mcsv1alpha1.ResourceImportSpec{
+			ClusterIDs: []string{common.LocalClusterID},
+			LabelIdentity: &mcsv1alpha1.LabelIdentitySpec{
+				Label: normalizedLabel3,
+				ID:    3,
+			},
+		},
+	}
+	existResImpList := &mcsv1alpha1.ResourceImportList{
+		Items: []mcsv1alpha1.ResourceImport{*labelIdentityResImp, *existingResImp},
+	}
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	fakeClient := fake.NewClientBuilder().WithScheme(common.TestScheme).WithLists(existResImpList).Build()
+	r := NewLabelIdentityExportReconciler(fakeClient, common.TestScheme, common.LeaderNamespace)
+	r.hashToLabels = map[string]string{
+		labelHash:  normalizedLabel,
+		labelHash2: normalizedLabel2,
+	}
+	// Mock the state of the reconciler where id 3 was allocated for labelHash3
+	r.labelsToID.Store(labelHash3, uint32(3))
+	r.allocator.setAllocated(3)
+	// The three keys should be drawn out of the queue almost concurrently after the worker starts
+	r.labelQueue.Add(labelHash)
+	r.labelQueue.Add(labelHash2)
+	r.labelQueue.Add(labelHash3)
+	go r.Run(stopCh)
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Using `time.Sleep` will introduce flake. Maybe we could use `WaitGroup` like below:
```
var wg sync.WaitGroup
wg.Add(3)
go func() {
    defer wg.Done()
    assert.NoError(t, c.syncLabelResourceImport(labelHash))
}()
go func() {
    defer wg.Done()
    assert.NoError(t, c.syncLabelResourceImport(labelHash2))
}()
go func() {
    defer wg.Done()
    assert.NoError(t, c.syncLabelResourceImport(labelHash3))
}()
wg.Wait()
```

Reply:
I would still prefer the queue workers to take out the keys and sync them , so that we have a better chance of simulating the concurrent processing. Nevertheless I changed the wait mechanism to be deterministic.